core-bundle/src/Intl/Locales.php

```diff
@@ -0,0 +1,312 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Intl;
+
+use Contao\CoreBundle\Framework\ContaoFramework;
+use Contao\System;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+class Locales
+{
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @var RequestStack
+     */
+    private $requestStack;
+
+    /**
+     * @var ContaoFramework
+     */
+    private $contaoFramework;
+
+    /**
+     * @var string
+     */
+    private $defaultLocale;
+
+    /**
+     * @var array<string>
+     */
+    private $locales;
+
+    /**
+     * @var array<string>
+     */
+    private $enabledLocales;
+
+    public function __construct(TranslatorInterface $translator, RequestStack $requestStack, ContaoFramework $contaoFramework, array $availableLocales, array $translatedLocales, string $defaultLocale, array $locales, array $enabledLocales)
```

Comment:
`array $translatedLocales, string $defaultLocale, array $locales, array $enabledLocales`

Really? What is the difference between all these arrays?

Reply:
Renamed in 3a8c19096ff111e95298cdba81070a9dfb4c51da to:  
`array $defaultLocales, array $defaultEnabledLocales, array $configLocales, array $configEnabledLocales`  
better?